### PR TITLE
Turn the CDDL test into a cabal test-suite and run test in the CI

### PR DESCRIPTION
**Pull request by MarcFontaine**
Part 1: Turn CDDL test into conditional cabal test-suite. To run the tests set the flag  `cddl`
Part 2: Run the CDDL test as part of the CI.

### CODE REVIEW

File: nix/pkgs.nix [hunk 1]
```diff
@@ -35,7 +35,8 @@ let
         packages.typed-protocols.configureFlags = [ "--ghc-option=-Werror" ];
         packages.io-sim.configureFlags = [ "--ghc-option=-Werror" ];
         packages.io-sim-classes.configureFlags = [ "--ghc-option=-Werror" ];
-        packages.ouroboros-network.configureFlags = [ "--ghc-option=-Werror" ];
+        packages.ouroboros-network.configureFlags = [ "--ghc-option=-Werror" "--flags=cddl"];
+        packages.ouroboros-network.components.tests.cddl.build-tools = [pkgs.cddl pkgs.cbor-diag];
```
Comment (MarcFontaine) [inline nix/pkgs.nix]: @rvl I have a cabal test suite called cddl that needs external Ruby tools from the `cddl` package and the `cbor-diag` package.
Is this the right way to add the packages to the CI ?

Comment (rvl) [reply nix/pkgs.nix]: Hi @MarcFontaine, yes this is exactly how you would add executable dependencies in `haskell.nix`.

File: nix/pkgs.nix [hunk 2]
```diff
@@ -35,7 +35,8 @@ let
         packages.typed-protocols.configureFlags = [ "--ghc-option=-Werror" ];
         packages.io-sim.configureFlags = [ "--ghc-option=-Werror" ];
         packages.io-sim-classes.configureFlags = [ "--ghc-option=-Werror" ];
-        packages.ouroboros-network.configureFlags = [ "--ghc-option=-Werror" ];
+        packages.ouroboros-network.configureFlags = [ "--ghc-option=-Werror" "--flags=cddl"];
```
Comment (rvl) [inline nix/pkgs.nix]: You could also set
```nix
packages.ouroboros-network.flags.cddl = true;
```

Comment (MarcFontaine) [reply nix/pkgs.nix]: OK changed that

**Comment by MarcFontaine** [comment]
> @MarcFontaine This looks great; Could you move the test to `ouroboros-network/tests` directory.
OK

**Pull request merged**